From 3a69933ed91fc413adb07a19d49cf19f9c36b894 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Thu, 13 Aug 2015 16:39:06 +0800
Subject: [PATCH 0196/5242] MLK-11344-11: spi: spi-imx: enable dma support for
 i.mx6dl

commit  a547c3f95433c717e1c1f8ac78a42a6a62869d08 from
https://source.codeaurora.org/external/imx/linux-imx.git

There is official workaound for TKT238285, so remove the limitation
for i.mx6dl.

Signed-off-by: Robin Gong <b38343@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/spi/spi-imx.c |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/spi/spi-imx.c b/drivers/spi/spi-imx.c
index d3d9bb8..3ba497c 100644
--- a/drivers/spi/spi-imx.c
+++ b/drivers/spi/spi-imx.c
@@ -1178,10 +1178,6 @@ static int spi_imx_sdma_init(struct device *dev, struct spi_imx_data *spi_imx,
 {
 	int ret;
 
-	/* use pio mode for i.mx6dl chip TKT238285 */
-	if (of_machine_is_compatible("fsl,imx6dl"))
-		return 0;
-
 	spi_imx->wml = spi_imx->devtype_data->fifo_size / 2;
 
 	/* Prepare for TX DMA: */
-- 
1.7.9.5

